File: pjt_tournois_e_sport_boot/src/main/java/projet/sopra/pjt_tournois_e_sport_boot/model/Journee.java
package projet.sopra.pjt_tournois_e_sport_boot.model;

import java.time.LocalDateTime;
import java.util.List;
import java.util.Objects;
import java.util.Set;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.EnumType;
import javax.persistence.Enumerated;
import javax.persistence.ForeignKey;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.OneToMany;
import javax.persistence.SequenceGenerator;
import javax.persistence.Table;
import javax.persistence.Transient;

import com.fasterxml.jackson.annotation.JsonView;

@Entity
@Table(name = "journee")
@SequenceGenerator(name = "seqJournee", sequenceName = "seq_journee", initialValue = 100, allocationSize = 1)

public class Journee {

	/// ATTRIBUTES

	@Id
	@GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "seqJournee")
	@JsonView({Views.JourneeWithTournoiAndMatch.class,Views.MatchWithJourneeAndResultat.class})
	private Long id;

	@ManyToOne
	@JoinColumn(name = "journee_tournoi", foreignKey = @ForeignKey(name = "journee_tournoi_fk"))
	@JsonView(Views.JourneeWithTournoiAndMatch.class)
	private Tournoi tournoi;

	@OneToMany(mappedBy = "journee")
	@JsonView(Views.JourneeWithTournoiAndMatch.class)
	private List<Match> matchsAJouerPourJournee;
//	@FutureOrPresent
	@Transient
	@Column(name = "journee_date_debut")
	@JsonView(Views.JourneeWithTournoiAndMatch.class)
	private LocalDateTime dateDebutJournee;
//	@Future
	@Transient
	@Column(name = "journee_date_fin")
	@JsonView(Views.JourneeWithTournoiAndMatch.class)
	private LocalDateTime dateFinJournee;
	@Enumerated(EnumType.STRING)
	@Column(name = "journee_etape")
	@JsonView(Views.JourneeWithTournoiAndMatch.class)
	private Etape etape;
	
	/// attribut pour connaitre l'ordre des journées d'une ligue/poule sans passer par les dates 
	@Column(name="journee_numero")
	@JsonView(Views.JourneeWithTournoiAndMatch.class)
	private int numero; 
	
	
	
	/// CONSTRUCTOR

	public Journee() {

	}

	public Journee(Tournoi tournoi, LocalDateTime dateDebutJournee, LocalDateTime dateFinJournee, Etape etape) {
		super();
		this.tournoi = tournoi;
		this.dateDebutJournee = dateDebutJournee;
		this.dateFinJournee = dateFinJournee;
		this.etape = etape;
	}
	
	public Journee(Tournoi tournoi, LocalDateTime dateDebutJournee, LocalDateTime dateFinJournee, Etape etape, int numero) {
		super();
		this.tournoi = tournoi;
		this.dateDebutJournee = dateDebutJournee;
		this.dateFinJournee = dateFinJournee;
		this.etape = etape;
		this.numero = numero; 
	}

	/// GETTERS

	public Long getId() {
		return id;
	}

	public Tournoi getTournoi() {
		return tournoi;
	}

	public List<Match> getMatchsAJouerPourJournee() {
		return matchsAJouerPourJournee;
	}

	public LocalDateTime getDateDebutJournee() {
		return dateDebutJournee;
	}

	public LocalDateTime getDateFinJournee() {
		return dateFinJournee;
	}

	public Etape getEtape() {
		return etape;
	}
	
	public int getNumero() {
		return numero;
	}

	/// SETTERS

	public void setId(Long idJournee) {
		this.id = idJournee;
	}

	public void setTournoi(Tournoi tournoi) {
		this.tournoi = tournoi;
	}

	public void setMatchsAJouerPourJournee(List<Match> matchsAJouerPourJournee) {
		this.matchsAJouerPourJournee = matchsAJouerPourJournee;
	}

	public void setDateDebutJournee(LocalDateTime dateDebutJournee) {
		this.dateDebutJournee = dateDebutJournee;
	}

	public void setDateFinJournee(LocalDateTime dateFinJournee) {
		this.dateFinJournee = dateFinJournee;
	}

	public void setEtape(Etape etape) {
		this.etape = etape;
	}

	public void setNumero(int numero) {
		this.numero = numero;
	}
	
	/// METHODS

	

	@Override
	public int hashCode() {
		return Objects.hash(id);
	}

	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		Journee other = (Journee) obj;
		return Objects.equals(id, other.id);
	}


}

File: pjt_tournois_e_sport_angular/src/app/model/journee.ts
import { identifierName } from '@angular/compiler';
import { Etape } from './etape.enum';
import { Tournoi } from './tournoi';
import { Match } from './match';
export class Journee {
  private _id: number | undefined;
  private _tournoi: Tournoi | undefined;
  private _matchsAJouerPourJournee: Match[] | undefined;
  private _etape: Etape | undefined;
  private _numero: number | undefined;

  constructor(
    id?: number,
    tournoi?: Tournoi,
    matchsAJouerPourJournee?: Match[],
    etape?: Etape,
    numero?: number
  ) {
    this._id = id;
    this._tournoi = tournoi;
    this._matchsAJouerPourJournee = matchsAJouerPourJournee;
    this._etape = etape;
    this._numero = numero;
  }

  public get id(): number | undefined {
    return this._id;
  }
  public set id(value: number | undefined) {
    this._id = value;
  }

  public get tournoi(): Tournoi | undefined {
    return this._tournoi;
  }
  public set tournoi(value: Tournoi | undefined) {
    this._tournoi = value;
  }

  public get matchsAJouerPourJournee(): Match[] | undefined {
    return this._matchsAJouerPourJournee;
  }
  public set matchsAJouerPourJournee(value: Match[] | undefined) {
    this._matchsAJouerPourJournee = value;
  }

  public get etape(): Etape | undefined {
    return this._etape;
  }
  public set etape(value: Etape | undefined) {
    this._etape = value;
  }

  public get numero(): number | undefined {
    return this._numero;
  }
  public set numero(value: number | undefined) {
    this._numero = value;
  }
}
